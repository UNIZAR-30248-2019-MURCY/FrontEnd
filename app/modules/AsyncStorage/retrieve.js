import { AsyncStorage } from 'react-native';

export const retrieveItem = async (key) => {
    let data ='';
    try {
        data = await AsyncStorage.getItem(key)
        console.log(data)
    } catch (error) {
        console.log(error.message);
    }
    return data;
}
