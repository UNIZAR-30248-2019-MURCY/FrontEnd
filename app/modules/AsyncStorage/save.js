import { AsyncStorage } from 'react-native';

export const saveData = async (key, data) => {
    try {
        await AsyncStorage.setItem(key, JSON.stringify(data));
        console.log('data stored');
    } catch (error) {
        // Error retrieving data
        console.log(error.message);
    }
};
