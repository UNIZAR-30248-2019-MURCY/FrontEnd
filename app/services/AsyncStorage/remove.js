import { AsyncStorage } from 'react-native';

export const removeItem = async (key) => {
    try {
        await AsyncStorage.removeItem(key);
        console.log('data removed');
        return true;
    } catch (error) {
        console.log(error.message);
        return false;
    }
}
