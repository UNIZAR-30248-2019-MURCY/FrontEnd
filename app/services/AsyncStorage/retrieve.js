import { AsyncStorage } from 'react-native';

export const retrieveItem = async (key) => {
    let data ='';
    try {
        let dataStored = await AsyncStorage.getItem(key)
        data = JSON.parse(dataStored).token;
        console.log('retrievedItem')
    } catch (error) {
        console.log(error.message);
    }
    return data;
}
