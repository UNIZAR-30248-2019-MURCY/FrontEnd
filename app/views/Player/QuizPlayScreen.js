import React, {Component} from 'react';
import {
    View,
    StyleSheet, Platform, ActivityIndicator,
} from 'react-native'
import {Button, colors, Text} from 'react-native-elements';
import Carousel from 'react-native-snap-carousel';
import {sliderWidth, itemWidth} from '../../modules/QuizzesCards/SliderEntry.style';
import SliderEntryQuestion from '../../components/SliderEntryQuestion';
import {ENTRIES1, ENTRIES2} from '../../modules/QuizzesCards/entries';
import Icon from 'react-native-vector-icons/FontAwesome';
import { scrollInterpolators, animatedStyles } from '../../modules/QuizzesCards/animations';
import {retrieveItem} from "../../modules/AsyncStorage/retrieve";
import {getQuizId, quizSearch} from "../../services/quiz/quizFuncs";

const WEB = Platform.OS === 'web';

export default class QuizPlayScreen extends Component {

    constructor(props) {
        super(props);
        this.state = {
            title:'',
            id: '',
            data:{
                questions:[]
            }
        }
    }

    componentDidMount() {
        if (this.props.navigation) {
            if(this.props.navigation.getParam('title')){
                this.setState({title: this.props.navigation.getParam('title')})

                this.setState({loading: true})
                retrieveItem('token')
                    .then(data => {
                        this.setState({token: JSON.parse(data)})
                        getQuizId(this.state.token, this.props.navigation.getParam('id'))
                            .then((data) => {
                                console.log(data);
                                this.setState({data: data});
                                this.setState({loading: false})
                            })
                            .catch((error) => {
                                this.setState({error: error.message})
                                this.setState({loading: false})
                            })
                    })
            }
        }
    }

    __renderItem({item, index}) {
        return <SliderEntryQuestion data={item} even={(index + 1) % 2 === 0}/>;
    }


    render() {
        const { color } = this.props;

        let showLoading = (
            this.state.loading ?
                <View style={[styles.containerLoading]} className='loadingShow'>
                    <ActivityIndicator animating={this.state.loading} size="large" color="grey"/>
                </View> :
                <View></View>
        );

        let buttons = (
            WEB ?
                <View style={styles.containerButton}>
                    <View style={styles.buttonContainer}>
                        <Button icon={
                            <Icon
                                name="arrow-left"
                                size={18}
                                color="white"
                            />
                        }
                                buttonStyle={styles.buttonPrevNext}
                                className='add-button'
                                onPress={() => {
                                    this.refs.carousel.snapToPrev();
                                }}/>
                    </View>
                    <View style={styles.buttonContainer}>
                        <Button icon={
                            <Icon
                                name="arrow-right"
                                size={18}
                                color="white"
                            />
                        }
                                buttonStyle={styles.buttonPrevNext}
                                className='remove-button'
                                onPress={() => {
                                    this.refs.carousel.snapToNext();
                                }}/>
                    </View>
                </View>:
                <View></View>
        );

        return (
            <View style={styles.containerUp}>
                <View style={styles.cross}>
                    <Button
                        className='close-button'
                        type="clear"
                        icon={
                            <Icon
                                name="times"
                                size={30}
                                color="grey"
                            />
                        }
                        onPress={() => {
                            this.props.navigation.goBack()
                        }
                        }
                    />
                </View>
                <View style={styles.container}>

                <View style={styles.containerTitle}>
                    <Text h3 style={{color: 'white', fontWeight: 'bold'}} numberOfLines={2} >{this.state.title}</Text>
                </View>
                    {showLoading}

                <Carousel
                    ref={'carousel'}
                    data={this.state.data.questions}
                    renderItem={this.__renderItem}
                    sliderWidth={sliderWidth}
                    itemWidth={itemWidth}
                    layout={'stack'}
                    scrollInterpolator={scrollInterpolators[`scrollInterpolator1`] }
                    slideInterpolatedStyle={ animatedStyles[`animatedStyles1`] }
                />
                {buttons}

                <View style={styles.sendContainer}>
                    <Button icon={
                        <Icon
                            name="paper-plane"
                            size={22}
                            color="white"
                        />
                    }
                            buttonStyle={styles.buttonSend}
                            className='remove-button'
                            onPress={() => {
                                this.refs.carousel.snapToNext();
                            }}/>
                </View>
                </View>
            </View>
        )
    }
}

const styles = StyleSheet.create({
    cross: {
        marginTop: 30,
        marginBottom: 0,
        marginRight: 30,
        alignItems: 'flex-end',
    },
    containerUp: {
        flex: 1,
        //justifyContent: 'center',
        backgroundColor: 'black'
    },
    container: {
        flex: 1,
        //justifyContent: 'center',
        alignItems: 'center',
        padding: 20,
        backgroundColor: 'black'
    },
    containerTitle: {
        marginTop: 0,
        marginBottom: 10,
        paddingRight: 20,
        paddingLeft: 20,
        paddingBottom: 20,
    },

    containerSettings: {
        padding: 20
    },
    button: {
        width: 200,
        height: 50,
        marginTop: 30,
        margin: 10,
        backgroundColor: 'grey',
        borderRadius: 14
    },
    buttonText: {
        fontSize: 20,
        color: 'grey'
    },
    buttonSO: {
        position: 'absolute',
        bottom: 0,
        marginBottom: 50,
    },
    containerButton: {
        flexDirection: 'row',
        alignItems: 'center',
        justifyContent: 'center',
    },
    buttonContainer: {
        alignItems: 'center',
        justifyContent: 'center',
    },
    sendContainer: {
        marginBottom: WEB ? 7 : 30,
        alignItems: 'flex-end',
        justifyContent: 'flex-end',
    },
    buttonPrevNext: {
        width: 120,
        height: 35,
        marginTop: 10,
        margin: 10,
        backgroundColor: 'black',
        borderRadius: 14,
        borderWidth: 0.5,
        borderColor: 'white'
    },
    buttonSend: {
        width: 60,
        height: 40,
        marginTop: 10,
        margin: 10,
        backgroundColor: 'black',
        borderRadius: 14,
        borderWidth: 0.5,
        borderColor: 'white'
    },


})
