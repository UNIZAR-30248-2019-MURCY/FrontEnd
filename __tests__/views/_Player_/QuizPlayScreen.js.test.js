import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import QuizPlayScreen from "../../../app/views/Player/QuizPlayScreen";
import {retrieveItem} from "../../../app/modules/AsyncStorage/retrieve";


describe('<QuizPlayScreen/>', () => {
    it('QuizPlayScreen renders without crashing', async () => {
        const rendered = renderer.create(<QuizPlayScreen/>).toJSON();
        expect(rendered).toBeTruthy();
    });
    it('QuizPlayScreen test against snapshot', async () => {
        await retrieveItem();
        const quizPlayScreen = renderer.create(<QuizPlayScreen />).toJSON();
        expect(quizPlayScreen).toMatchSnapshot();
    });
    it('should render the Loading component if state.loading is true', async () => {
        await retrieveItem();

        let createComponent = shallow(<QuizPlayScreen />);
        createComponent.setState({ loading: true });
        expect(createComponent.find('.loadingShow').length).toBe(1);
    });
    it('should not render the Loading component if state.loading is false', async () => {
        await retrieveItem();

        let createComponent = shallow(<QuizPlayScreen />);
        createComponent.setState({ loading: false });
        expect(createComponent.find('.loadingShow').length).toBe(0);
    });
    it('Replace and navigate to Points', async () => {
        await retrieveItem();

        const navigationMock = { replace: jest.fn(),
            getParam: jest.fn() };
        const wrapper = shallow(<QuizPlayScreen navigation={navigationMock} />);
        wrapper.setState({
            points:[]
        })
        wrapper.find('.send-button').simulate('press');
        expect(navigationMock.replace.mock.calls.length).toEqual(1);
        expect(navigationMock.replace.mock.calls[0]).toEqual(['Points',{points: 0}]);
    });
});
