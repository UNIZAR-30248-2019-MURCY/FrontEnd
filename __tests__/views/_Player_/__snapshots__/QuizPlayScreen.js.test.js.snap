// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<QuizPlayScreen/> QuizPlayScreen test against snapshot 1`] = `
<View
  style={
    Object {
      "backgroundColor": "black",
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "flex-end",
        "marginBottom": 0,
        "marginRight": 30,
        "marginTop": 30,
      }
    }
  >
    <View
      style={
        Object {
          "borderRadius": 3,
        }
      }
    >
      <View
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "transparent",
              "borderColor": "#2089dc",
              "borderRadius": 3,
              "borderWidth": 0,
              "flexDirection": "row",
              "justifyContent": "center",
              "padding": 8,
            }
          }
        >
          <Text
            allowFontScaling={false}
            style={
              Array [
                Object {
                  "color": "grey",
                  "fontSize": 30,
                },
                undefined,
                Object {
                  "fontFamily": "FontAwesome",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
                Object {},
              ]
            }
          >
            
          </Text>
        </View>
      </View>
    </View>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "black",
        "flex": 1,
        "padding": 20,
      }
    }
  >
    <View
      style={
        Object {
          "marginBottom": 10,
          "marginTop": 0,
          "paddingBottom": 20,
          "paddingLeft": 20,
          "paddingRight": 20,
        }
      }
    >
      <Text
        numberOfLines={2}
        style={
          Object {
            "color": "white",
            "fontSize": 35,
            "fontWeight": "bold",
          }
        }
        theme={
          Object {
            "colors": Object {
              "disabled": "hsl(208, 8%, 90%)",
              "divider": "#bcbbc1",
              "error": "#ff190c",
              "grey0": "#393e42",
              "grey1": "#43484d",
              "grey2": "#5e6977",
              "grey3": "#86939e",
              "grey4": "#bdc6cf",
              "grey5": "#e1e8ee",
              "greyOutline": "#bbb",
              "platform": Object {
                "android": Object {
                  "error": "#f44336",
                  "primary": "#2196f3",
                  "secondary": "#9C27B0",
                  "success": "#4caf50",
                  "warning": "#ffeb3b",
                },
                "ios": Object {
                  "error": "#ff3b30",
                  "primary": "#007aff",
                  "secondary": "#5856d6",
                  "success": "#4cd964",
                  "warning": "#ffcc00",
                },
              },
              "primary": "#2089dc",
              "searchBg": "#303337",
              "secondary": "#8F0CE8",
              "success": "#52c41a",
              "warning": "#faad14",
            },
          }
        }
      >
        
      </Text>
    </View>
    <View />
    <RCTScrollView
      activeAnimationOptions={null}
      activeAnimationType="timing"
      activeSlideAlignment="center"
      activeSlideOffset={20}
      apparitionDelay={0}
      automaticallyAdjustContentInsets={false}
      autoplay={false}
      autoplayDelay={1000}
      autoplayInterval={3000}
      callbackOffsetMargin={5}
      containerCustomStyle={Object {}}
      contentContainerCustomStyle={Object {}}
      contentContainerStyle={
        Array [
          Object {
            "paddingLeft": 67.5,
            "paddingRight": 67.5,
          },
          Object {},
        ]
      }
      data={Array []}
      decelerationRate="fast"
      directionalLockEnabled={true}
      enableMomentum={false}
      enableSnap={true}
      firstItem={0}
      hasParallaxImages={false}
      horizontal={true}
      inactiveSlideOpacity={0.7}
      inactiveSlideScale={0.9}
      inactiveSlideShift={0}
      inverted={false}
      itemWidth={615}
      layout="stack"
      lockScrollTimeoutDuration={1000}
      lockScrollWhileSnapping={false}
      loop={false}
      loopClonesPerSide={3}
      onLayout={[Function]}
      onResponderRelease={[Function]}
      onScroll={[Function]}
      onScrollBeginDrag={[Function]}
      onScrollEndDrag={[Function]}
      onStartShouldSetResponderCapture={[Function]}
      onTouchEnd={[Function]}
      onTouchStart={[Function]}
      overScrollMode="never"
      pinchGestureEnabled={false}
      removeClippedSubviews={false}
      renderItem={[Function]}
      scrollEnabled={true}
      scrollEventThrottle={1}
      scrollInterpolator={[Function]}
      scrollsToTop={false}
      shouldOptimizeUpdates={true}
      showsHorizontalScrollIndicator={false}
      showsVerticalScrollIndicator={false}
      slideInterpolatedStyle={[Function]}
      slideStyle={Object {}}
      sliderWidth={750}
      style={
        Object {
          "flexDirection": "row",
          "opacity": 0,
          "width": 750,
        }
      }
      swipeThreshold={20}
      useScrollView={false}
      vertical={false}
    >
      <View />
    </RCTScrollView>
    <View />
    <View
      style={
        Object {
          "alignItems": "flex-end",
          "justifyContent": "flex-end",
          "marginBottom": 40,
        }
      }
    >
      <View
        style={
          Object {
            "borderRadius": 14,
          }
        }
      >
        <View
          accessible={true}
          isTVSelectable={true}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "white",
                "borderColor": "white",
                "borderRadius": 14,
                "borderWidth": 0.5,
                "flexDirection": "row",
                "height": 40,
                "justifyContent": "center",
                "margin": 10,
                "marginTop": 10,
                "padding": 8,
                "width": 60,
              }
            }
          >
            <Text
              allowFontScaling={false}
              style={
                Array [
                  Object {
                    "color": "black",
                    "fontSize": 22,
                  },
                  undefined,
                  Object {
                    "fontFamily": "FontAwesome",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                  Object {},
                ]
              }
            >
              
            </Text>
          </View>
        </View>
      </View>
      <Text
        numberOfLines={2}
        style={
          Object {
            "alignSelf": "center",
            "color": "white",
            "fontSize": 15,
          }
        }
        theme={
          Object {
            "colors": Object {
              "disabled": "hsl(208, 8%, 90%)",
              "divider": "#bcbbc1",
              "error": "#ff190c",
              "grey0": "#393e42",
              "grey1": "#43484d",
              "grey2": "#5e6977",
              "grey3": "#86939e",
              "grey4": "#bdc6cf",
              "grey5": "#e1e8ee",
              "greyOutline": "#bbb",
              "platform": Object {
                "android": Object {
                  "error": "#f44336",
                  "primary": "#2196f3",
                  "secondary": "#9C27B0",
                  "success": "#4caf50",
                  "warning": "#ffeb3b",
                },
                "ios": Object {
                  "error": "#ff3b30",
                  "primary": "#007aff",
                  "secondary": "#5856d6",
                  "success": "#4cd964",
                  "warning": "#ffcc00",
                },
              },
              "primary": "#2089dc",
              "searchBg": "#303337",
              "secondary": "#8F0CE8",
              "success": "#52c41a",
              "warning": "#faad14",
            },
          }
        }
      >
        Send quiz
      </Text>
    </View>
  </View>
</View>
`;
